File: api-commons/src/main/java/com/github/alexeysol/geekregime/apicommons/models/dtos/users/UserDto.java
package com.github.alexeysol.geekregime.apicommons.models.dtos.users;

import com.github.alexeysol.geekregime.apicommons.models.utils.HasId;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDto implements HasId {
    private long id;
    private String email;
    private Date createdAt;
    private Date updatedAt;
    private UserDetailsDto details;
}

File: src/main/kotlin/com/github/alexeysol/geekregimeapiusers/models/dtos/UserDto.kt
package com.github.alexeysol.geekregimeapiusers.models.dtos

import java.time.Instant

data class UserDto(
    var id: Long = 0L,
    var email: String = "",
    var createdAt: Instant = Instant.now(),
    var updatedAt: Instant = Instant.now(),
    var details: DetailsDto? = null,
)
